File: gotourl.js
console.log('\nACTION!\n');
const button = document.getElementById("clickGO")


function getRandom(num) {
  console.log('\nRANDOM GO\n');
  return Math.floor(Math.random() * (num + 1));
}

function goto(){
  console.log('\nGOTO\n');

  const fileURL = 'styles_list.txt';
  const xhr = new XMLHttpRequest();
  xhr.open('GET', fileURL, true);
  xhr.onreadystatechange = function() {
    if (xhr.readyState === 4 && xhr.status === 200) {
      const content = xhr.responseText;
      const lines = content.split('\n');
      const index = getRandom(lines.length);
      const line = lines[index];
      console.log(index);
  
      const url = `https://www.midlibrary.io/styles/${line}`;
      console.log(url);
    
      window.open(url, '_blank');
    }
  };
  xhr.send();
}

button.addEventListener('click', function() {goto()});

File: gotourl.py
"""
Open a page with random artstyle
"""

import random
import os


def get_random(num):
    return random.randint(0, num)


with open("styles_list.txt", "r") as file:
    content = file.readlines()
    index = get_random(len(content))
    line = content[index]
    print(index)

url = f'https://www.midlibrary.io/styles/{line}'
print(url)

try:
    os.system("start " + str(url))
except Exception as ex:
    print(ex)


